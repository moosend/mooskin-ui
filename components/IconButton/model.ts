import { IBoxComponentProps } from '../Box/model';

export interface IIconButtonComponentProps extends IBoxComponentProps {
	/** provide to make the button disabled */
	disabled?: boolean;
}
