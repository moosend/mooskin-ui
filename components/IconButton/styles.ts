import styled from 'styled-components';

// Models
import { IIconButtonComponentProps } from './model';

// Components
import { Box } from '../Box/Box';

// "CSS" variables
import variables from '../_utils/globals/variables';

export const StyledIconButton = styled(Box)<IIconButtonComponentProps>`
	font-family: 'Mooskin Icons';
	font-size: 21px;
	font-style: normal;
	color: ${(props) => props.palette?.fontColors.primary2 || variables.fontColors.primary2};
	text-decoration: none;
	cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
	opacity: ${(props) => (props.disabled ? 0.5 : 1)};
`;

StyledIconButton.displayName = 'StyledIconButton';
