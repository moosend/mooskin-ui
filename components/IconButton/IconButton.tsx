import * as React from 'react';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Models
import { IIconButtonComponentProps } from './model';

// Styled Components
import { StyledIconButton } from './styles';

/**
 * IconButton
 */
export const IconButton: React.FC<IIconButtonComponentProps> = withMooskinContext((props) => {
	const onClick = (e: React.MouseEvent<HTMLElement>) => {
		!props.disabled && props.onClick && props.onClick(e);
	};
	return <StyledIconButton {...props} onClick={onClick} className={`notranslate ${props.className}`} />;
});

IconButton.displayName = 'IconButton';

IconButton.defaultProps = {
	className: '',
	style: {}
};
