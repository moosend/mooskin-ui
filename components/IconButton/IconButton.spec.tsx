import * as React from 'react';
import { IconButton } from './IconButton';

import { mount } from 'enzyme';

describe('SmallIconButton', () => {
	test('renders correctly', () => {
		const func = jest.fn();

		const tree = mount(<IconButton onClick={func}>close</IconButton>);
		expect(tree).toMatchSnapshot();
	});

	test('callback is called when clicked', () => {
		const func = jest.fn();

		const component = mount(<IconButton onClick={func}>close</IconButton>);

		component.find('StyledIconButton').simulate('click');
		expect(func).toHaveBeenCalled();
	});

	test('renders a disabled button', () => {
		const func = jest.fn();

		const component = mount(
			<IconButton onClick={func} disabled>
				close
			</IconButton>
		);

		component.find('StyledIconButton').simulate('click');
		expect(func).not.toHaveBeenCalled();
	});
});
