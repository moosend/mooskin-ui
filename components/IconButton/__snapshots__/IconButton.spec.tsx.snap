// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallIconButton renders correctly 1`] = `
<WithMooskinContext
  onClick={[MockFunction]}
>
  <Component
    onClick={[MockFunction]}
    palette={
      Object {
        "backgroundColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
  >
    <StyledIconButton
      onClick={[Function]}
      palette={
        Object {
          "backgroundColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "borderColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "fontColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
        }
      }
    >
      <Box
        className="sc-gtssRu fehwhe"
        onClick={[Function]}
        palette={
          Object {
            "backgroundColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "borderColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "fontColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
          }
        }
        style={Object {}}
        transition="0.2s all"
      >
        <Component
          className="sc-gtssRu fehwhe"
          onClick={[Function]}
          palette={
            Object {
              "backgroundColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "borderColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "fontColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
            }
          }
          style={Object {}}
          transition="0.2s all"
        >
          <StyledBox
            className="sc-gtssRu fehwhe"
            onClick={[Function]}
            palette={
              Object {
                "backgroundColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "borderColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "fontColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
              }
            }
            style={Object {}}
            transition="0.2s all"
          >
            <div
              className="sc-bdnylx jknATe sc-gtssRu fehwhe"
              onClick={[Function]}
              style={Object {}}
            >
              close
            </div>
          </StyledBox>
        </Component>
      </Box>
    </StyledIconButton>
  </Component>
</WithMooskinContext>
`;
